tests/integration_tests/test_shap.py

```diff
@@ -71,3 +77,78 @@ def test_mean_abs_shap_importance_evaluator_with_target() -> None:
     )
 
     assert param_importance != param_importance_with_target
+
+
+@pytest.mark.parametrize("inf_value", [float("inf"), -float("inf")])
+def test_shap_importance_evaluator_with_infinite(inf_value: float) -> None:
+    # The test ensures that trials with infinite values are ignored to calculate importance scores.
+    n_trial = 10
+    seed = 13
+
+    # Importance scores are calculated without a trial with an inf value.
+    study = create_study(sampler=RandomSampler(seed=seed))
+    study.optimize(objective, n_trials=n_trial)
+
+    evaluator = ShapleyImportanceEvaluator(seed=seed)
+    param_importance_without_inf = evaluator.evaluate(study)
+
+    # A trial with an inf value is added into the study manually.
+    study.add_trial(
+        create_trial(
+            value=inf_value,
+            params={"x1": 1.0, "x2": 1.0, "x3": 3.0},
+            distributions={
+                "x1": FloatDistribution(low=0.1, high=3),
+                "x2": FloatDistribution(low=0.1, high=3, log=True),
+                "x3": FloatDistribution(low=2, high=4, log=True),
+            },
+        )
+    )
+    # Importance scores are calculated with a trial with an inf value.
+    param_importance_with_inf = evaluator.evaluate(study)
```

Comment:
```suggestion
    param_importance_with_inf = get_param_importances(study, evaluator=evaluator)
```
Maybe this is out of scope of this PR, but I think we should use public interfaces in tests whenever it is possible.

Reply:
Thanks. How about applying the change as a follow-up when we unify the test as suggested in https://github.com/optuna/optuna/pull/3576#pullrequestreview-981549111 after we merge https://github.com/optuna/optuna/pull/3552?

Reply:
Since as you said the suggested change is supposed to be out-of-scope in terms of the aim of this PR.